drivers/input/pinnacle/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources(pinnacle.c)
+zephyr_library_sources(pinnacle_spi.c)
```

Comment:
Cool, one more file organization change before getting to the code, could you merge the whole thing back into a single file (including the header) and drop the device directory? The one-per-directory organization is common in sensor because it's common practice to split the trigger stuff out but this can now happily live in a single file and makes code maintenance a bit easier.

For the bus access functions (the SPI stuff), I know this sensor is commonly used in I2C mode as well, would you be able to add the function call struct for it and split the binding already? Then stuff would be in place to add i2c mode as well. Bonus point for adding the i2c variant too, it should be simpler than the SPI, but can also be done in a followup.

Thanks!